sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -874,13 +874,29 @@ class SessionCatalog(
     // the extra columns that we don't require), with UpCast (to make sure the type change is
     // safe) and Alias (to respect user-specified view column names) according to the view schema
     // in the catalog.
-    val projectList = viewColumnNames.zip(metadata.schema).map { case (name, field) =>
-      Alias(UpCast(UnresolvedAttribute.quoted(name), field.dataType), field.name)(
-        explicitMetadata = Some(field.metadata))
+    val projectList = if (isTempView) {
+      viewColumnNames.zip(metadata.schema).map { case (name, field) =>
+        Alias(UpCast(UnresolvedAttribute.quoted(name), field.dataType), field.name)(
+          explicitMetadata = Some(field.metadata))
+      }
+    } else {
+      viewColumnNames.zip(metadata.schema).
+        map { case (name, field) => innerStruct(Seq(), name, field)}
     }
     View(desc = metadata, isTempView = isTempView, child = Project(projectList, parsedPlan))
   }
 
+  private def innerStruct(parent : Seq[String], name : String,
+                          field : StructField) : NamedExpression = {
+    field.dataType match {
+      case structType : StructType => Alias(CreateStruct.create(structType.map {
```

Comment:
For most cases that the references table is not altered, doesn't this produce unnecessary `CreateStruct`?

Reply:
Maybe but i dit not know if this making a huge difference, and i do not found a way to check at this moment of the analysis of the code  if the references table is altered or not.